docs/guide/ethereum-provider.md

```diff
@@ -1,103 +1,164 @@
+Please read the [Upcoming Breaking Changes section](#upcoming-breaking-changes) for details.
+:::
 
-[[toc]]
+The provider API is specified by [EIP-1193](https://eips.ethereum.org/EIPS/eip-1193), and is designed to be minimal.
+We recommend that all web3 site developers read the [Basic Usage section](#basic-usage).
 
-## Upcoming Provider Changes
+## Table of Contents
 
-In **Q4 2020**, we will be making some breaking changes to the provider API. The biggest change is that we will stop injecting `window.web3`. We encourage you to
-[read more about this here](https://github.com/MetaMask/metamask-extension/issues/8077).
+[[toc]]
 
-## List of Chain and Network IDs
+## Upcoming Breaking Changes
 
 ::: tip
-The chain and network IDs of these chains are the same.
+These changes are _upcoming._ Follow [this GitHub issue](https://github.com/MetaMask/metamask-extension/issues/8077) for details.
 
-Network IDs are decimal strings, while chain IDs are hexadecimal strings.
+If you are new to using the provider, or use [ethers](https://www.npmjs.com/package/ethers), you do not have to worry about these changes, and can skip ahead [to the next section](#api).
 :::
 
-```javascript
-'1': Ethereum Main Network
-'2': Morden Test network
-'3': Ropsten Test Network
-'4': Rinkeby Test Network
-'5': Goerli Test Network
-'42': Kovan Test Network
-```
+### `window.ethereum` API Changes
 
-## Properties
+In **Q3 2020** (date TBD), we are introducing some breaking changes to this API, which we encourage you to
+[read more about here](https://medium.com/metamask/breaking-changes-to-the-metamask-inpage-provider-b4dde069dd0a).
 
-### ethereum.isMetaMask
+At that time, we will:
 
-Returns `true` or `false`, representing whether the user has MetaMask installed.
+- Stop emitting `chainIdChanged`, and instead emit `chainChanged`
+- Ensure that all `chainId` values are **not** 0-prefixed
+  - For example, instead of `0x01`, we will always return `0x1`, wherever the chain ID is returned or accessible.
+- Remove the following experimental methods:
+  - `ethereum._metamask.isEnabled`
+  - `ethereum._metamask.isApproved`
 
-### ethereum.autoRefreshOnNetworkChange
+These changes _may_ break your website.
+Please read our [migration guide](about:blank) for more details.
 
-When the network is changed, MetaMask will reload any pages that have made requests to the provider.
-This automatic reload behavior will be removed in a future release of MetaMask, but in the meantime it can be disabled with this flag.
+### `window.web3` Removal
 
-To disable auto-refresh on a network change you can do:
+::: tip
+If you do not use the `window.web3` object injected by MetaMask, you will not be affected by these changes.
+:::
 
-```javascript
-ethereum.autoRefreshOnNetworkChange = false;
-```
+In **Q4 2020** (date TBD), we will:
 
-This can be toggled on or off at any time.
+- Stop injecting `window.web3` into web pages
+- Remove the `ethereum.autoRefreshOnNetworkChange` property
```

Comment:
Are we definitely removing this? I don't think we should mention this until we're prepared to commit to removing this, and we have a plan on how to replace it (e.g. no more automatic refreshes, or a replacement property, etc.)

This leaves me confused about what I'm supposed to do about this removal. Our current error messages strongly advise that all dapps set this property after all.

Reply:
> Our current error messages strongly advise that all dapps set this property after all.

For the time being, yes, but given that we're removing the object whose behavior the property controls, I don't see how we can keep the property around. I think it'll be confusing if it starts to control the behavior of `window.ethereum` once `window.web3` is removed.

What to do instead can be mentioned in the migration guide. We can have another go at discussing auto-reload today.

Reply:
It doesn't seem quite accurate to frame the automatic reload as being something the `window.web3` property _does_. I don't see this property as "controlling the behavior of `window.web3`".

I agree that it is confusing to say that we're going to remove `window.web3` without specifying how the automatic reload behavior will change. But still - better to leave the information out if we're not sure yet. Saying we'll remove the property and then deciding to keep it would be _even more_ confusing after all.

I think it is a legitimate possibility that we'll keep the property if we're keeping the automatic reload behavior itself (even if the behavior does change somewhat). We can rename it, but we'd keep the old name around for backwards-compatibility, because... why not. The only scenario where we'd want to completely remove it is if we don't do automatic page reloading anymore at all. Which... seems likely? But we should finalize that decision first in any case. Hopefully we can do that soon.

Reply:
I'll remove any references to removal for now and we can resume this discussion for the migration guide.